File: include/webengine/libqwebenginefullscreenrequest.zig
const C = @import("qt6c");

/// https://doc.qt.io/qt-6/qwebenginefullscreenrequest.html
pub const qwebenginefullscreenrequest = struct {
    /// New constructs a new QWebEngineFullScreenRequest object.
    ///
    /// ``` other: ?*C.QWebEngineFullScreenRequest ```
    pub fn New(other: ?*anyopaque) ?*C.QWebEngineFullScreenRequest {
        return C.QWebEngineFullScreenRequest_new(@ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qwebenginefullscreenrequest.html#operator=)
    ///
    /// ``` self: ?*C.QWebEngineFullScreenRequest, other: ?*C.QWebEngineFullScreenRequest ```
    pub fn OperatorAssign(self: ?*anyopaque, other: ?*anyopaque) void {
        C.QWebEngineFullScreenRequest_OperatorAssign(@ptrCast(self), @ptrCast(other));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qwebenginefullscreenrequest.html#reject)
    ///
    /// ``` self: ?*C.QWebEngineFullScreenRequest ```
    pub fn Reject(self: ?*anyopaque) void {
        C.QWebEngineFullScreenRequest_Reject(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qwebenginefullscreenrequest.html#accept)
    ///
    /// ``` self: ?*C.QWebEngineFullScreenRequest ```
    pub fn Accept(self: ?*anyopaque) void {
        C.QWebEngineFullScreenRequest_Accept(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qwebenginefullscreenrequest.html#toggleOn)
    ///
    /// ``` self: ?*C.QWebEngineFullScreenRequest ```
    pub fn ToggleOn(self: ?*anyopaque) bool {
        return C.QWebEngineFullScreenRequest_ToggleOn(@ptrCast(self));
    }

    /// [Qt documentation](https://doc.qt.io/qt-6/qwebenginefullscreenrequest.html#origin)
    ///
    /// ``` self: ?*C.QWebEngineFullScreenRequest ```
    pub fn Origin(self: ?*anyopaque) ?*C.QUrl {
        return C.QWebEngineFullScreenRequest_Origin(@ptrCast(self));
    }

    /// Delete this object from C++ memory.
    ///
    /// ``` self: ?*C.QWebEngineFullScreenRequest ```
    pub fn QDelete(self: ?*anyopaque) void {
        C.QWebEngineFullScreenRequest_Delete(@ptrCast(self));
    }
};

File: src_original/webengine/libqwebenginefullscreenrequest.cpp
#include <QUrl>
#include <QWebEngineFullScreenRequest>
#include <qwebenginefullscreenrequest.h>
#include "libqwebenginefullscreenrequest.h"
#include "libqwebenginefullscreenrequest.hxx"

QWebEngineFullScreenRequest* QWebEngineFullScreenRequest_new(const QWebEngineFullScreenRequest* other) {
    return new QWebEngineFullScreenRequest(*other);
}

void QWebEngineFullScreenRequest_OperatorAssign(QWebEngineFullScreenRequest* self, const QWebEngineFullScreenRequest* other) {
    self->operator=(*other);
}

void QWebEngineFullScreenRequest_Reject(QWebEngineFullScreenRequest* self) {
    self->reject();
}

void QWebEngineFullScreenRequest_Accept(QWebEngineFullScreenRequest* self) {
    self->accept();
}

bool QWebEngineFullScreenRequest_ToggleOn(const QWebEngineFullScreenRequest* self) {
    return self->toggleOn();
}

QUrl* QWebEngineFullScreenRequest_Origin(const QWebEngineFullScreenRequest* self) {
    return new QUrl(self->origin());
}

void QWebEngineFullScreenRequest_Delete(QWebEngineFullScreenRequest* self) {
    delete self;
}
